### ISSUE TITLE
Support for NEO Extended Attributes

### ISSUE THREAD

**Issue by all5n** [mucommander/mucommander#194]
Any chance of support for NIO extended attributes via java.nio.file.attribute.UserDefinedFileAttributeView?

I was looking at adding it myself to your code.  Seems like it would need some sort of configuration object to allow the user to define the attribute name strings.  Then some sort of subclass of SimpleFileAttributes that could read the file attributes specified in the configuration and provide them as requested.

Then, somewhere in the UI, have the ability to add the new attributes as another column in the file grid view and file properties.

As the architects of the software, i would appreciate your input as to the best way to go about this.

I will even provide the results as a pull request you may do with as you will.

**Comment by ahadas** [comment]
Hi @all5n ,
Supporting extended file attributes sounds like a cool addition!

I tend to think that, at least at the beginning, we can simply display them in the file-properties dialog rather than also integrating them in the file tables (grids). Dynamically changing the file tables can be challenging.

Regarding extending SimpleFileAttribute, note that it is only used by subclasses of ArchiveEntry and SyncedFileAttributes. We would probably want to have them on every LocalFile as well, right? If so, that may not be the best design to extend SimpleFileAttributes.

Moreover, there was a request to support file-comments in macOS which are defined as external attribute as well (see #80). Seems that in this case, the comment is not part of UserDefinedFileAttributeView but we can get it using JNA:
```java
XAttrUtil.getXAttr(path, "com.apple.metadata:kMDItemFinderComment"));
```
It would be great to implement it in a way that we can enhance the additional attributes to support things like that comment and not tie it specifically with UserDefinedFileAttributeView.

Let me know if that makes sense.
I would love cooperating with you to get this in!

**Comment by all5n** [comment]
Hey, thanks for the response, and thanks for pointing me in the right direction.  I was pretty sure SimpleFileAttribute was not the right place as your model is somewhat elegant/sophisticated and i had not studied it in detail.

One of my requirements is that i need to be able to view the attribute easily on the grid without having to open another dialog/properties window.  I realize that Swing can be somewhat picky about adding columns dynamically, so i was thinking that i might add a couple extra "generic" columns that could be assigned to an attribute at runtime and are by default, hidden.  Then use some configuration element to be able to assign it to an appropriate NIO attribute (or XAttr file comment!).

Day job will be crummy for the next few weeks.  This is on the list.

**Comment by ahadas** [comment]
@all5n , We may need to think a bit more about the implementation at the UI, maybe the best way would be to implement a small POC that demonstrates how it will look like. But anyway, this sounds like a very good idea and I'm looking forward to seeing patches :) (and don't worry about not getting to this in the next few weeks). Please don't hesitate to approach me if you need any help in the design/implementation of this!

**Comment by JamesOlvertone** [comment]
Yes I planned this, but the code is not that easy to understand. It makes a lot diffs for each platform I did not expect this.

I discovered that for macosx my first idea is still implemented: showing the xattr./file comments if there are some in the pref dialog. 
First I thought, oh great, put the MacOSX version as a template for the linux version, change the relevant xattributes code, thats it. But thats not how it works here. The linux package diffs again for the three desktops: gnome, kde, xfce. There are some guessing classes, an Activator-class and a Bundle-Interface. Heres where I get stucked, where I think if they are even relevant for my task and where to put the implementation of the getExtendFileProperties(). 
Introduce a new Class thats used in the guessing classes for each desktop, ...? Any ideas?

**Comment by pskowronek** [comment]
Is the work still in progress?

**Comment by JamesOlvertone** [comment]
No its not. I moved to other projects with higher priority.
